### ISSUE TITLE
Cannot boot to boot.wim files of DLC1

### ISSUE THREAD

**Issue by steve6375** [ipxe/wimboot#31]
DLC Boot 2022 v4.0.220520.iso

e.g. \DLC1\WinPE\W10x64.wim

![image](https://user-images.githubusercontent.com/1469350/176631026-9e3c2374-801d-4dd5-9aaa-568e96b8a43b.png)

Using current wimboot binary

**Comment by NiKiZe** [comment]
As it says not supported compression, does this wim even work in standard winpe boot?

The fix for you is to repack the wim with standard supported compression.

**Comment by steve6375** [comment]
yes, it boots using bootmgr.

**Comment by NiKiZe** [comment]
Don't know what DLC1 is, or why you put it into an iso when you only need the wim. (Are you confusing wim as being an iso?)

Could you use wimlib-imagex info on the wim and post output (as text), see wimlib.net (available as chocolatey package if you are on windows)

**Comment by steve6375** [comment]
![image](https://user-images.githubusercontent.com/1469350/176661422-332e17b7-07a5-41a9-b237-a48377e9fbc9.png)

**Comment by steve6375** [comment]
http://dlcboot.com/download.html

**Comment by steve6375** [comment]
The wimboot in a1ive's grubfm project can boot the wim file.
But I cannot seem to load the grubfm version of the wimboot file in grub4dos using a grub4dos kernel command as I can with a standard wimboot binary for some reason.

**Comment by NiKiZe** [comment]
> The wimboot in a1ive's grubfm project can boot the wim file. But I cannot seem to load the grubfm version of the wimboot file in grub4dos using a grub4dos kernel command as I can with a standard wimboot binary for some reason.

Yes some Grub developer has done some weird stuff to clone parts of wimboot functionality instead of adding support into grub to not force Linux kernel boot for efi binaries with a kernel signature.

But I don't see why you bring that up here.

**Comment by steve6375** [comment]
v2.7.4 it works!!! Fantastic !!! Thank you so much.

**Comment by mcb30** [comment]
@a1ive Thanks for the patch - that was a lot simpler than I was expecting!  I've created #32 to add this support upstream.  If you let me know your name and email address for the commit message, I'll add an appropriate credit before merging.  Thanks again!

**Comment by a1ive** [comment]
> @a1ive Thanks for the patch - that was a lot simpler than I was expecting! I've created #32 to add this support upstream. If you let me know your name and email address for the commit message, I'll add an appropriate credit before merging. Thanks again!

`Signed-off-by: Li Gen <ligenlive@gmail.com>`
